### PR TITLE
[ACADEMIC-15949] New mocked summary_handler endpoint

### PR DESCRIPTION
**Pull request by rijuma**
# [ACADEMIC-15949](https://jira.2u.com/browse/ACADEMIC-15949)

Added mocked endpoint as a placeholder summary_handler to be used to fetch and process the summary from the database.

It also validates the call with `should_apply_to_block()` to check if it should be called.

Example of the hardcoded response:
```json
{
   "contentId":"some-content-uuid",
   "courseId":"some-course-uuid",
   "data":[
      {
         "id":"this-flashy-uuid",
         "type":"VIDEO",
         "text":"Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec arcu nulla, porttitor sed volutpat nec, eleifend venenatis leo. Ut luctus libero nisi. Nam elementum scelerisque purus in pretium. Etiam in interdum nibh, vel dictum ligula. Nunc orci nunc, consequat ut efficitur vitae, tincidunt non sapien. Integer id sollicitudin erat. Praesent egestas odio quis vulputate ornare. ",
         "created_at":"2023-05-18 19:12:14",
         "updated_at":"2023-05-18 19:12:14"
      },
      {
         "id":"this-stunning-uuid",
         "type":"CONTENT",
         "text":"Nunc dignissim dapibus lectus, a ultrices est tempus quis. Fusce congue lorem et urna tempor luctus. Phasellus tincidunt mauris at sodales facilisis. Nam tortor erat, porttitor ac laoreet vitae, molestie non lacus. Praesent eu fermentum lacus. Fusce lectus risus, sagittis ut justo in, vulputate sodales elit. In vitae tempor nulla. Phasellus tincidunt ante nec enim pharetra.",
         "created_at":"2023-05-18 19:12:14",
         "updated_at":"2023-05-18 19:12:14"
      }
   ],
   "created_at":"2023-05-18 19:12:14",
   "updated_at":"2023-05-18 19:12:14"
}
```

### CODE REVIEW

**Comment by ashultz0** [comment]
nice!

Since that's simple I think we might be able to fill in the top two dummy fields in this PR based on things the summary aside already knows or can get by figuring out the usage ID of the block it is attached to like rapid response does https://github.com/mitodl/rapid-response-xblock/blob/master/rapid_response_xblock/block.py#L214

**Comment by rijuma** [comment]
> nice!
> 
> Since that's simple I think we might be able to fill in the top two dummy fields in this PR based on things the summary aside already knows or can get by figuring out the usage ID of the block it is attached to like rapid response does https://github.com/mitodl/rapid-response-xblock/blob/master/rapid_response_xblock/block.py#L214

Updated!

File: ai_aside/summaryhook_aside/block.py [hunk 1]
```diff
@@ -54,6 +60,54 @@ class SummaryHookAside(XBlockAside):
     XBlock aside that injects AI summary javascript
     """
 
+    def _get_block(self):
+        """
+        Gets the current aside block.
```
Comment (ashultz0) [inline ai_aside/summaryhook_aside/block.py:65]: probably more accurate to say "gets the block wrapped by this aside"

File: ai_aside/summaryhook_aside/block.py [hunk 2]
```diff
@@ -54,6 +60,54 @@ class SummaryHookAside(XBlockAside):
...
+        block = self._get_block()
+        valid = self.should_apply_to_block(block)
+
+        if not valid:
+            return Response(status=404)
+
+        timestr = time.strftime('%Y-%m-%d %H:%M:%S')
+        json = {
+            "contentId": "some-content-uuid",
+            "courseId": "some-course-uuid",
```
Comment (ashultz0) [inline ai_aside/summaryhook_aside/block.py:86]: now that you have the block I think you can fill content ID and course ID just like how student_view_aside does

File: ai_aside/summaryhook_aside/block.py (hunk 1 above)
Comment (rijuma) [reply ai_aside/summaryhook_aside/block.py]: Correct. Updated.

**Review by ashultz0 — APPROVED** [review]

**Pull request merged**